src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormContent.jsx

```diff
@@ -377,54 +451,89 @@ const DataSeriesFormItem = ({
           id: dataItem.dataSourceId,
           content: {
             value: dataItem.label,
-            icon: isComplexDataSeries ? (
-              <div
-                style={{
-                  width: '1rem',
-                  height: '1rem',
-                  backgroundColor:
-                    dataItem.color ||
-                    DATAITEM_COLORS_OPTIONS[i % DATAITEM_COLORS_OPTIONS.length],
-                }}
-              />
-            ) : null,
+            icon:
+              cardConfig.type === CARD_TYPES.TIMESERIES ||
+              cardConfig.type === CARD_TYPES.BAR ? (
+                <div
+                  style={{
+                    width: '1rem',
+                    height: '1rem',
+                    backgroundColor:
+                      dataItem.color ||
+                      DATAITEM_COLORS_OPTIONS[
+                        i % DATAITEM_COLORS_OPTIONS.length
+                      ],
+                  }}
+                />
+              ) : null,
             rowActions: () => [
-              <Button
-                key={`data-item-${dataItem.dataSourceId}`}
-                renderIcon={isComplexDataSeries ? Subtract16 : Edit16}
-                hasIconOnly
-                kind="ghost"
-                size="small"
-                onClick={() => {
-                  if (isComplexDataSeries) {
-                    const filteredItems = cardConfig.content?.series?.filter(
-                      (item) => item.dataSourceId !== dataItem.dataSourceId
-                    );
-                    setSelectedDataItems(
-                      filteredItems.map((item) => item.dataSourceId)
+              !isComplexDataSeries && [
+                <Button
+                  key={`data-item-${dataItem.dataSourceId}_edit`}
+                  renderIcon={Edit16}
+                  hasIconOnly
+                  kind="ghost"
+                  size="small"
+                  onClick={() => {
```

Comment:
This seems like a lot of logic for an inline function. Can we pull out into a function that we pass?